docs/madr/decisions/009-tracing-policy.md

```diff
@@ -0,0 +1,221 @@
+configs_](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/trace/v3/http_tracer.proto#envoy-v3-api-msg-config-trace-v3-tracing-http).
+
+The rationale here is that the provider is likely to be the same provider for all
+traces of a `Mesh` so it makes sense to configure it on the `Mesh`.
+
+##### Further provider config options
+
+There are two more possibilities worth mentioning.
+
+We could easily offer defining a `Tracing`-specific backend inline:
+
+```yaml
+spec:
+  default:
+    backendConfig:
+      type: <backend type>
+      <type-specific config>
+```
+
+We could create an additional resource `TracingConfig` (name tbd) that can be referenced
+from `Tracing`:
+
+```yaml
+spec:
+  default:
+    configRef:
+      name: <resource-name>
+```
+
+#### Additional configuration
+
+This MADR proposes additional options be configurable on `Tracing` resources
+directly.
+
+##### Sampling
+
+At the moment, tracing backends only support Envoy's
+[`overall_sampling`](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#envoy-v3-api-field-extensions-filters-network-http-connection-manager-v3-httpconnectionmanager-tracing-overall-sampling)
+via the `sampling` field.
+
+Question: should we allow the following instead?
+
+```yaml
+spec:
+  default:
+    sampling:
+      overall: <percentage>
+      client: <percentage>
+      random: <percentage>
+```
+
+If users set the simple `sampling` field we have now,
+do they expect it to limit traces started via `x-client-trace-id`?
+Do they want to disable `x-client-trace-id`?
+
+##### Tags
+
+Envoy can set tags in traces using values from:
+
+- Literal value
+- Request header
+- Environment variable
+- Metadata
+
+We propose allowing users to configure custom tags using only literal values or
+request headers. Using Kuma leaves environment variables and Envoy metadata
+opaque for users so the use cases for configuring them (at least directly)
+in Kuma are limited.
+
+Tags can be configured as follows:
+
+```yaml
+spec:
+  default:
+    tags:
+      - name:
+        # exactly one of the following keys must be set
+        literal: ...
+        header:
+          name:
+          default:
+```
+
+##### Method/path specific
+
+Users can set the `spec.targetRef` to be a `MeshGatewayRoute` or
+`MeshTrafficRoute`.
+
+Both paths and methods are matchable in `MeshGatewayRoute` and we can ensure
+that `MeshTrafficRoute` also supports this matching.
+
+### Examples
+
+All examples assume Kubernetes.
+
+#### Simple
+
+This configures all `Dataplane` inbounds with `kuma.io/service: backend` to
+trace a maximum of 80% of traffic and adds the custom tag `team` with a value of
+`core` to spans.
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: TrafficLog
+metadata:
+  name: all
+  labels:
+    kuma.io/mesh: default
+spec:
+  targetRef:
+    kind: MeshService
+    name: backend
+  default:
+    backend: jaeger
```

Comment:
This should be consistent with `MeshAccessLog` [reference](https://github.com/kumahq/kuma/pull/4908/files#diff-c3dc29a3f68a72bc0b16dc3b13326c706a60a0d6c0675f2c7b5a82234de883cfR210-R213). I'm not saying one or the other name is better I'm just saying that we should have a consistent way of "referencing" things.

Reply:
@slonka you mean adopt the following from your MADR, right?

```
      default:
        backends:
          - type: tcp
            conf:
              address: 127.0.0.1:5000
          - type: reference
            conf: 
              kind: MeshAccessLogBackend
              name: file-backend
```